lua/nvim-web-devicons.lua

```diff
@@ -428,9 +481,15 @@ function M.get_icon_colors(name, ext, opts)
   local is_strict = if_nil(opts and opts.strict, global_opts.strict)
   local icon_data
   if is_strict then
-    icon_data = icons_by_filename[name] or get_icon_by_extension(name, ext, opts) or (has_default and default_icon)
+    icon_data = icons_by_filename[name]
+      or get_icon_by_prefix(name, pre_or_ext, opts)
+      or get_icon_by_extension(name, pre_or_ext, opts)
+      or (has_default and default_icon)
   else
-    icon_data = icons[name] or get_icon_by_extension(name, ext, opts) or (has_default and default_icon)
+    icon_data = icons[name]
+      or get_icon_by_prefix(name, pre_or_ext, opts)
```

Comment:
This will result in a prefix being returned when users were previously expecting an extension, when the key is the same.